sdk/src/main/java/tech/relaycorp/sdk/Message.kt

```diff
@@ -0,0 +1,57 @@
+package tech.relaycorp.sdk
+
+import java.time.ZonedDateTime
+import java.util.*
+
+abstract class Message(
+    val id: Message.Id,
+    val message: ByteArray,
+    senderEndpoint: Endpoint,
+    receiverEndpoint: Endpoint,
+    val creationDate: ZonedDateTime = ZonedDateTime.now(),
+    val expirationDate: ZonedDateTime = maxExpirationDate()
+) {
+    companion object {
+        internal fun maxExpirationDate() = ZonedDateTime.now().plusDays(30)
+    }
+
+    class Id
```

Comment:
I'd prefer to have top-level classes throughout this library: I think inner classes make the developer digress from the class/entity they're dealing with.